Mathlib/Data/Int/GCD.lean

```diff
@@ -511,255 +513,265 @@ theorem pow_gcd_eq_one {M : Type _} [Monoid M] (x : M) {m n : ℕ} (hm : x ^ m =
--- #align tactic.norm_num.prove_lcm_int tactic.norm_num.prove_lcm_int
-
--- /-- Evaluates the `nat.coprime` function. -/
--- unsafe def prove_coprime_nat (c : instance_cache) (ex ey : expr) :
---     tactic (instance_cache × Sum expr expr) := do
---   let x ← ex.toNat
---   let y ← ey.toNat
---   match x, y with
---     | 1, _ => pure (c, Sum.inl <| q(Nat.coprime_one_left).mk_app [ey])
---     | _, 1 => pure (c, Sum.inl <| q(Nat.coprime_one_right).mk_app [ex])
---     | 0, 0 => pure (c, Sum.inr q(Nat.not_coprime_zero_zero))
---     | 0, _ => do
---       let c ← mk_instance_cache q(ℕ)
---       let (c, p) ← prove_lt_nat c q(1) ey
---       pure (c, Sum.inr <| q(nat_coprime_helper_zero_left).mk_app [ey, p])
---     | _, 0 => do
---       let c ← mk_instance_cache q(ℕ)
---       let (c, p) ← prove_lt_nat c q(1) ex
---       pure (c, Sum.inr <| q(nat_coprime_helper_zero_right).mk_app [ex, p])
---     | _, _ => do
---       let c ← mk_instance_cache q(ℕ)
---       let (d, a, b) := Nat.xgcdAux x 1 0 y 0 1
---       if d = 1 then do
---           let (c, ea) ← c a
---           let (c, eb) ← c b
---           let (c, etx, px) ← prove_mul_nat c ex ea
---           let (c, ety, py) ← prove_mul_nat c ey eb
---           let (c, p) ← if a ≥ 0 then
---             prove_add_nat c ety q(1) etx else prove_add_nat c etx q(1) ety
---           let pf : expr := if a ≥ 0 then q(nat_coprime_helper_2) else q(nat_coprime_helper_1)
---           pure (c, Sum.inl <| pf [ex, ey, ea, eb, etx, ety, px, py, p])
---         else do
---           let (c, ed) ← c d
---           let (c, eu) ← c (x / d)
---           let (c, ev) ← c (y / d)
---           let (c, _, pu) ← prove_mul_nat c ed eu
---           let (c, _, pv) ← prove_mul_nat c ed ev
---           let (c, p) ← prove_lt_nat c q(1) ed
---           pure (c, Sum.inr <| q(nat_not_coprime_helper).mk_app [ed, ex, ey, eu, ev, pu, pv, p])
--- #align tactic.norm_num.prove_coprime_nat tactic.norm_num.prove_coprime_nat
-
--- /-- Evaluates the `gcd`, `lcm`, and `coprime` functions. -/
--- @[norm_num]
--- unsafe def eval_gcd : expr → tactic (expr × expr)
---   | q(Nat.gcd $(ex) $(ey)) => do
---     let c ← mk_instance_cache q(ℕ)
---     Prod.snd <$> prove_gcd_nat c ex ey
---   | q(Nat.lcm $(ex) $(ey)) => do
---     let c ← mk_instance_cache q(ℕ)
---     Prod.snd <$> prove_lcm_nat c ex ey
---   | q(Nat.Coprime $(ex) $(ey)) => do
---     let c ← mk_instance_cache q(ℕ)
---     prove_coprime_nat c ex ey >>= Sum.elim true_intro false_intro ∘ Prod.snd
---   | q(Int.gcd $(ex) $(ey)) => do
---     let zc ← mk_instance_cache q(ℤ)
---     let nc ← mk_instance_cache q(ℕ)
---     (Prod.snd ∘ Prod.snd) <$> prove_gcd_int zc nc ex ey
---   | q(Int.lcm $(ex) $(ey)) => do
---     let zc ← mk_instance_cache q(ℤ)
---     let nc ← mk_instance_cache q(ℕ)
---     (Prod.snd ∘ Prod.snd) <$> prove_lcm_int zc nc ex ey
---   | _ => failed
--- #align tactic.norm_num.eval_gcd tactic.norm_num.eval_gcd
-
--- end NormNum
-
--- end Tactic
+namespace Tactic
```

Comment:
I think we should move this to `Mathlib.Tactic.NormNum.GCD`, interleaving it with the math was probably not a good idea.

Reply:
Moved